kornia/augmentation/container/image.py

```diff
@@ -116,17 +119,121 @@ def __init__(
             ), f"Expect a tuple of (int, int). Got {self.random_apply}."
         else:
             self.random_apply = False
+        self._validate_mix_augmentation(*args, validate_args=validate_args)
+        self.has_mix_augmentation = len(self.__get_mix_indices__(iter(args))) > 0
+
+    def update_attribute(
+        self,
+        same_on_batch: Optional[bool] = None,
+        return_transform: Optional[bool] = None,
+        keepdim: Optional[bool] = None,
+    ) -> None:
+        for arg in self.children():
+            if isinstance(arg, _AugmentationBase):
+                if same_on_batch is not None:
+                    arg.same_on_batch = same_on_batch
+                if return_transform is not None:
+                    arg.return_transform = return_transform
+                if keepdim is not None:
+                    arg.keepdim = keepdim
+
+    @property
+    def same_on_batch(self) -> Optional[bool]:
+        return self._same_on_batch
+
+    @same_on_batch.setter
+    def same_on_batch(self, same_on_batch: Optional[bool]) -> None:
+        self._same_on_batch = same_on_batch
+        self.update_attribute(same_on_batch=same_on_batch)
+
+    @property
+    def return_transform(self) -> Optional[bool]:
+        return self._return_transform
+
+    @return_transform.setter
+    def return_transform(self, return_transform: Optional[bool]) -> None:
+        self._return_transform = return_transform
+        self.update_attribute(return_transform=return_transform)
+
+    @property
+    def keepdim(self) -> Optional[bool]:
+        return self._keepdim
+
+    @keepdim.setter
+    def keepdim(self, keepdim: Optional[bool]) -> None:
+        self._keepdim = keepdim
+        self.update_attribute(keepdim=keepdim)
+
+    def clear_state(self) -> None:
+        self._params = []
+
+    def _validate_mix_augmentation(self, *args: nn.Module, validate_args: bool = True):
+        mix_count = 0
+        for arg in args:
+            if isinstance(arg, (MixAugmentationBase,)):
+                mix_count += 1
+        if self.random_apply is False and mix_count > 1 and validate_args:
+            raise ValueError(
+                f"Multiple mix augmentation is prohibited without enabling random_apply. Detected {mix_count}."
+            )
+        if mix_count > 1 and validate_args:
+            warnings.warn(
+                f"Multiple ({mix_count}) mix augmentation detected and at most one mix augmenation can"
+                "be applied at each forward. To silence this warning, please set `validate_args` to False.",
+                category=UserWarning,
+            )
+
+    def __get_mix_indices__(self, args: Iterator) -> List[int]:
```

Comment:
could you add at least one line function definition?

Reply:
I will refactor those mess soon. Pretty ugly now.